src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs

```diff
@@ -189,12 +189,12 @@ protected Task<AuthenticationTicket> AuthenticateOnceAsync()
             }
         }
 
-        protected abstract Task<AuthenticationTicket> AuthenticateAsync();
+        protected abstract Task<AuthenticationTicket> HandleAuthenticateAsync();
 
         public bool ShouldHandleScheme(string authenticationScheme)
         {
             return string.Equals(BaseOptions.AuthenticationScheme, authenticationScheme, StringComparison.Ordinal) ||
-                (BaseOptions.AutomaticAuthentication && string.IsNullOrWhiteSpace(authenticationScheme));
+                (BaseOptions.AutomaticAuthentication && authenticationScheme == string.Empty);
```

Comment:
Brent brought up an issue that we use automatic with an empty string never null. I am on the fence. 

But I am fine with treating null and empty the same as we were before if you feel strongly about it.